src/libstd/macros.rs

```diff
@@ -213,6 +213,27 @@ macro_rules! eprintln {
     ($fmt:expr, $($arg:tt)*) => (eprint!(concat!($fmt, "\n"), $($arg)*));
 }
 
+#[macro_export]
+#[unstable(feature = "await_macro", issue = "50547")]
+#[allow_internal_unstable]
+#[allow_internal_unsafe]
+macro_rules! await {
+    ($e:expr) => { {
+        let mut pinned = $e;
+        let mut pinned = unsafe { ::core::mem::PinMut::new_unchecked(&mut pinned) };
+        loop {
+            match ::std::raw::with_get_cx(|cx|
+                      ::core::future::Future::poll(pinned.reborrow(), cx))
```

Comment:
Doesn't this macro need `$crate`? Or have the rules changed, @petrochenkov?

Reply:
It doesn't absolutely need it unless it's used from std itself (because of the extern prelude), but I'll make the change.

Reply:
Done.